### PR TITLE
Make CL calculations not update accumulators

### PR DESCRIPTION
**Pull request by ValarDragon**
Make CL CalcOutAmtGivenIn and CalcInAmtGivenOut skip fetching and setting the accumulators. This saves notably on the CPU and I/O time at hand here. This should also help smart contracts.

At our last benchmark, this should be saving .75% of sync speed, but is of much higher impact with all our v24 work conjoined. (sync speed saving all coming from protorev) This also paves the way for a second PR removing the CacheCtx on `CalcOutAmtGivenIn` which would save another 1% at current benchmarks. (Also from protorev)

## Summary by CodeRabbit

- **New Features**
	- Improved speed and gas efficiency in CL price estimations by preventing unnecessary state writes.

- **Refactor**
	- Updated calculation functions in concentrated liquidity to support conditional accumulator updates, enhancing operational efficiency.

### CODE REVIEW

File: x/concentrated-liquidity/swaps.go [hunk 1]
```diff
@@ -652,29 +663,33 @@ func (k Keeper) swapCrossTickLogic(ctx sdk.Context,
 	nextInitializedTick int64, nextTickIter db.Iterator,
 	p types.ConcentratedPoolExtension,
 	spreadRewardAccum *accum.AccumulatorObject, uptimeAccums *[]*accum.AccumulatorObject,
-	tokenInDenom string) (SwapState, error) {
+	tokenInDenom string, updateAccumulators bool) (SwapState, error) {
```
Comment (ValarDragon) [inline x/concentrated-liquidity/swaps.go:666]: in a subsequent PR I think I'll package these accum vars into their own struct.

File: x/concentrated-liquidity/swaps.go [hunk 2]
```diff
@@ -652,29 +663,33 @@ func (k Keeper) swapCrossTickLogic(ctx sdk.Context,
...
+		}
 
-	// Retrieve the liquidity held in the next closest initialized tick
-	spreadRewardGrowth := sdk.DecCoin{Denom: tokenInDenom, Amount: swapState.globalSpreadRewardGrowthPerUnitLiquidity}
-	liquidityNet, err := k.crossTick(ctx, p.GetId(), nextInitializedTick, &nextInitializedTickInfo, spreadRewardGrowth, spreadRewardAccum.GetValue(), *uptimeAccums)
-	if err != nil {
-		return swapState, err
+		// Retrieve the liquidity held in the next closest initialized tick
+		spreadRewardGrowth := sdk.DecCoin{Denom: tokenInDenom, Amount: swapState.globalSpreadRewardGrowthPerUnitLiquidity}
+		_, err := k.crossTick(ctx, p.GetId(), nextInitializedTick, &nextInitializedTickInfo, spreadRewardGrowth, spreadRewardAccum.GetValue(), *uptimeAccums)
```
Comment (ValarDragon) [inline x/concentrated-liquidity/swaps.go:687]: In a subsequent PR I'll refactor crossTick to not return LiquidityNet, it just reads that from one of the args.

**Review by czarcas7ic — APPROVED** [review]
Nice find!

**Comment by czarcas7ic** [comment]
This should have a changelog

**Review by p0mvn — APPROVED** [review]
LGTM

**Pull request merged**